test/unit/process/test_cuda_process_set.cpp

```diff
@@ -63,25 +70,37 @@ void testRandomSystemAddForcingTerms(std::size_t n_cells, std::size_t n_reaction
     processes.push_back(micm::Process::create().reactants(reactants).products(products).phase(gas_phase));
   }
 
-  micm::ProcessSet cpu_set{ processes, state.variable_map_ };
-  micm::CudaProcessSet gpu_set{ processes, state.variable_map_ };
+  micm::ProcessSet cpu_set{ processes, cpu_state.variable_map_ };
+  micm::CudaProcessSet gpu_set{ processes, gpu_state.variable_map_ };
 
-  for (auto& elem : state.variables_.AsVector())
+  for (auto& elem : cpu_state.variables_.AsVector())
     elem = get_double();
+  for (int i = 0; i < cpu_state.variables_.AsVector().size(); i++)
+  {
+    gpu_state.variables_.AsVector()[i] = cpu_state.variables_.AsVector()[i];
+  }
+  gpu_state.variables_.CopyToDevice();
 
-  MatrixPolicy<double> rate_constants{ n_cells, n_reactions };
-  for (auto& elem : rate_constants.AsVector())
+  CPUMatrixPolicy<double> cpu_rate_constants{ n_cells, n_reactions };
+  GPUMatrixPolicy<double> gpu_rate_constants{ n_cells, n_reactions };
+  for (auto& elem : cpu_rate_constants.AsVector())
     elem = get_double();
```

Comment:
```suggestion
  std::ranges::generate(cpu_rate_constants.AsVector(), get_double);
```
Again, the for loop is perfectly fine but this might have other benefits behind the scenes using the generator.

Reply:
Done.